lambda_functions/stac/stac.py

```diff
@@ -78,19 +79,30 @@ def get_bucket_and_key(message):
 
 def stac_dataset(metadata_doc, item_abs_path, parent_abs_path):
     """
-    Convert a YAML dict to a STAC Item dict
+    Returns a dict corresponding to a stac item catalog
     """
 
-    # TODO: Check projection before converting! Not everything is EPSG:3577
-    geodata = valid_coord_to_geojson(metadata_doc['grid_spatial']
-                                     ['projection']['valid_data']
-                                     ['coordinates'])
+    if metadata_doc['grid_spatial']['projection'].get('valid_data', None):
```

Comment:
This code would be much easier to read by adding a new variable like `gs_proj = metadata_doc['grid_spatial']['projection']`

The geometry handling code could even be a separate function.